pkg/sdk/poc/generator/helper_methods.go

```diff
@@ -0,0 +1,115 @@
+package generator
+
+import (
+	"fmt"
+	"log"
+)
+
+type objectIdentifier string
+
+const (
+	AccountObjectIdentifier             objectIdentifier = "AccountObjectIdentifier"
+	DatabaseObjectIdentifier            objectIdentifier = "DatabaseObjectIdentifier"
+	SchemaObjectIdentifier              objectIdentifier = "SchemaObjectIdentifier"
+	SchemaObjectIdentifierWithArguments objectIdentifier = "SchemaObjectIdentifierWithArguments"
+)
+
+func identifierStringToObjectIdentifier(s string) objectIdentifier {
+	switch s {
+	case "AccountObjectIdentifier":
+		return AccountObjectIdentifier
+	case "DatabaseObjectIdentifier":
+		return DatabaseObjectIdentifier
+	case "SchemaObjectIdentifier":
+		return SchemaObjectIdentifier
+	case "SchemaObjectIdentifierWithArguments":
+		return SchemaObjectIdentifierWithArguments
+	default:
+		return ""
+	}
+}
+
+type HelperMethod struct {
+	Name        string
+	StructName  string
+	ReturnValue string
+	ReturnType  string
+}
+
+func newHelperMethod(name, structName, returnValue string, returnType string) *HelperMethod {
+	return &HelperMethod{
+		Name:        name,
+		StructName:  structName,
+		ReturnValue: returnValue,
+		ReturnType:  returnType,
+	}
+}
+
+var requiredFieldsForIDMethodMapping map[objectIdentifier][]string = map[objectIdentifier][]string{
+	AccountObjectIdentifier:  {"Name"},
+	DatabaseObjectIdentifier: {"Name", "DatabaseName"},
+	SchemaObjectIdentifier:   {"Name", "DatabaseName", "SchemaName"},
+}
+
+func newIDHelperMethod(structName string, objectIdentifier objectIdentifier) *HelperMethod {
+	var args string
+	fields := requiredFieldsForIDMethodMapping[objectIdentifier]
+	for _, field := range fields {
+		args += fmt.Sprintf("v.%v, ", field)
+	}
+	returnValue := fmt.Sprintf("New%v(%v)", objectIdentifier, args)
+	return newHelperMethod("ID", structName, returnValue, string(objectIdentifier))
+}
+
+func newObjectTypeHelperMethod(structName string) *HelperMethod {
+	returnValue := fmt.Sprintf("ObjectType%v", structName)
+	return newHelperMethod("ObjectType", structName, returnValue, "ObjectType")
+}
+
+func containsFieldNames(fields []*Field, names ...string) bool {
+	fieldNames := map[string]any{}
+	for _, field := range fields {
+		fieldNames[field.Name] = nil
+	}
+
+	for _, name := range names {
+		if _, ok := fieldNames[name]; !ok {
```

Comment:
You can use `slices.ContainsFunc`. IMO it's more readable.

Reply:
done